RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,613 @@
+elements from zero value to any value, including an indefinite
+fill value as a future extension.
+
+## Motivation and Scope
+
+In general, the unspecified elements of sparse tensors have
+domain-specific interpretations:
+- In Linear Algebra domain, the unspecified elements are zero-valued.
+- In Graph domain, the unspecified elements of adjacency matrices
+  represent non-edges. In this document, a non-edge corresponds to
+  an unspecified fill value.
+- In neural networks, sparse tensors can be inputs, say, to activation
+  functions that are defined in terms of elementary functions.
+  Evaluation of such functions on sparse tensors can be element-wise,
+  or one-dimension-wise operations. This includes the evaluation of
+  the functions on the values corresponding to unspecified
+  elements. Even when the unspecified elements are initially
+  zero-valued, the values of unspecified elements may be mapped to
+  nonzero values (see [an example
+  below](#application-random-sequence-of-rare-events-with-non-zero-mean)). So,
+  in the domain of Calculus, the unspecified elements of sparse
+  tensors can have any defined value.
+
+Currently, in PyTorch, the unspecified elements of sparse tensors are
+zero-valued with few exceptions. For instance, the
+`torch.sparse.softmax` function assumes that the unspecified elements
+of a sparse tensor input are negative infinity (-inf) valued.
+
+In PyTorch 1.6, element-wise functions from Calculus such as `exp`,
+`log`, etc, or arithmetic operations such as addition, subtraction,
+etc, on sparse tensors are not supported, because the existing
+functions on sparse tensors assume that the unspecified elements are
+zero valued (this applies to functions in `torch` namespace, functions
+in `torch.sparse` namespace may use a different interpretation for
+unspecified elements).
+
+To support Calculus functions on sparse tensors, we propose adding a
+`fill_value` property to PyTorch sparse tensors that will represent
+the values of unspecified sparse tensor elements.  While doing so, we
+also need to consider:
+- how the nonzero fill value affects the result of linear algebra
+  operations such as matrix multiplication, etc.
+- how to define a fill value for hybrid tensors (tensors with sparse
+  and dense dimensions),
+- how PyTorch autograd support needs to deal with a specified
+  fill value.
+
+
+## Proposal
+
+This proposal is about enabling sparse tensor support for the Calculus
+domain while preserving the existing functionality for the Linear
+Algebra domain, and allowing extensions to the Graph domain.
+
+In the following, we describe the sparse tensor `fill_value` feature
+with examples involving sparse tensors in COO storage format.  When a
+new sparse tensor storage format is introduced to PyTorch, the same
+semantics should be applied to the new format.
+
+0.  Used terminology
+
+    - "Defined value" is a value for which memory is allocated and
+      initialized.
```

Comment:
```suggestion
      initialized to some defined value.
```

Reply:
applied without "defined" to avoid tautology.